gbrl/src/cpp/optimizer.cpp

```
@@ -94,6 +107,35 @@ Optimizer* Optimizer::loadFromFile(std::ifstream& file){
+    int start_idx = this->start_idx, stop_idx = this->stop_idx;
+    float lr = this->scheduler->get_lr(t);
+#ifndef _MSC_VER
+    #pragma omp simd
+#endif
+    for (int i = start_idx; i < stop_idx; i++){
+        scaled_grad_theta[i] = -lr * raw_grad_theta[i];
+    }
+    
+}
+
+void Optimizer::add_scaled(float *raw_grad_theta, const float *scaled_grad_theta, int t){
```

Cppcheck (2.19.0)

[style] The function 'add_scaled' is never used.

(unusedFunction)